core/src/main/java/net/sourceforge/jnlp/util/OptionsDefinitionsPrinter.java

```diff
@@ -0,0 +1,149 @@
+module.  An independent module is a module which is not derived from
+or based on this library.  If you modify this library, you may extend
+this exception to your version of the library, but you are not
+obligated to do so.  If you do not wish to do so, delete this
+exception statement from your version.
+*/
+package net.sourceforge.jnlp.util;
+
+import net.sourceforge.jnlp.util.docprovider.TextsProvider;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static net.adoptopenjdk.icedteaweb.option.OptionsDefinitions.OPTIONS;
+
+/**
+ * This class allows to print the various set of options.
+ */
+public class OptionsDefinitionsPrinter {
+
+    public static List<OPTIONS> getItwsettingsCommands() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.HELP2,
+            OPTIONS.LIST,
+            OPTIONS.GET,
+            OPTIONS.INFO,
+            OPTIONS.SET,
+            OPTIONS.RESET,
+            OPTIONS.RESETALL,
+            OPTIONS.HEADLESS,
+            OPTIONS.CHECK,
+            OPTIONS.VERBOSE
+        });
+    }
+
+    public static List<OPTIONS> getPolicyEditorOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.HELP1,
+            OPTIONS.FILE,
+            OPTIONS.DEFAULTFILE,
+            OPTIONS.CODEBASE,
+            OPTIONS.SIGNEDBY,
+            OPTIONS.PRINCIPALS,
+            OPTIONS.VERBOSE
+            }
+        );
+    }
+
+    public static List<OPTIONS> getJavaWsControlOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.ABOUT,
+            OPTIONS.VIEWER,
+            OPTIONS.CLEARCACHE,
+            OPTIONS.LISTCACHEIDS,
+            OPTIONS.LICENSE,
+            OPTIONS.HELP1}
+        );
+    }
+
+    public static List<OPTIONS> getJavaWsRuntimeOptions() {
+        return Arrays.asList(new OPTIONS[]{
+            OPTIONS.VERSION,
+            OPTIONS.ARG,
+            OPTIONS.PARAM,
+            OPTIONS.PROPERTY,
+            OPTIONS.UPDATE,
+            OPTIONS.VERBOSE,
+            OPTIONS.NOSEC,
+            OPTIONS.NOUPDATE,
+            OPTIONS.HEADLESS,
+            OPTIONS.STRICT,
+            OPTIONS.XML,
+            OPTIONS.REDIRECT,
+            OPTIONS.NOFORK,
+            OPTIONS.NOHEADERS,
+            OPTIONS.OFFLINE,
+            OPTIONS.TRUSTNONE,
+            OPTIONS.JNLP,
+            OPTIONS.HTML,
+            OPTIONS.BROWSER
+        });
+    }
+
+    public static List<OPTIONS> getJavaWsOptions() {
+        final List<OPTIONS> l = new ArrayList<>();
+        l.addAll(getJavaWsRuntimeOptions());
+        l.addAll(getJavaWsControlOptions());
+        //trustall is not returned by getJavaWsRuntimeOptions
+        //or getJavaWsControlOptions, as it is not desired in documentation
+        l.add(OPTIONS.TRUSTALL);
+        return l;
+    }
+
+    public static void main(String[] args) throws IOException {
```

Comment:
check that args is not null and has size > 0

Reply:
done